### PR TITLE
Tests for WebAppNameContextInitializer

### PR DESCRIPTION
**Pull request by littleaj**
I noticed this was missing tests. They're pretty straightforward.

I did optimize the filter test class a bit: using mockito instead of class override to give it more flexibility.

### CODE REVIEW

**Comment by SergeyKanzhelev** [comment]
I doubt Java SDK is better then .NET in handling slot swap. If it is - add test case for it. See https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/648 for details

**Comment by SergeyKanzhelev** [comment]
@littleaj which is exactly the problem with the slot swap. As I said .NET is not better than that now... we entertained an idea to periodically check the environment variable for update which probably what we will end up implementing. Perhaps add ignored test for now =)

**Comment by dhaval24** [comment]
@SergeyKanzhelev could a better way be to have some sort of callback when the slot change happens? Checking of environment variable would still have some delay between the when slot change happens and SDK picks up. Till then App Map would surface incorrect information I believe thoughts ?

File: web/src/test/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilterTests.java [hunk 1]
```diff
@@ -74,7 +64,7 @@ public void testFilterInitializedSuccessfullyFromConfiguration() throws ServletE
         Filter filter = createInitializedFilter();
         WebModulesContainer container = ServletUtils.getWebModuleContainer(filter);
 
-        Assert.assertNotNull("Container shouldn't be null", container);
+        assertNotNull("Container shouldn't be null", container);
         Assert.assertTrue("Modules container shouldn't be empty", container.getModulesCount() > 0);
```
Comment (dhaval24) [inline web/src/test/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilterTests.java]: I would like consistency if possible, can we stay consistent in importing asserts. Here Assert.assertFalse is for example imported and Assert.assertTrue isn't.

File: web/src/test/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilterTests.java [hunk 2]
```diff
@@ -156,6 +146,18 @@ public void testUnhandledIOExceptionWithTelemetryClientThatThrows() throws Illeg
...
+        FilterChain chain = mock(FilterChain.class);
+
+        mocks.filter.doFilter(ServletUtils.generateDummyServletRequest(), ServletUtils.generateDummyServletResponse(), chain);
+        String role = mocks.mockTelemetryClient.getContext().getCloud().getRole();
+        assertNotNull(role);
+        assertEquals(this.mockContextPath.replace("/", ""), role);
+
+    }
+
     // region Private methods
```
Comment (dhaval24) [inline web/src/test/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilterTests.java]: This kind of C# stuff also we should consider cleaning :)

**Comment by SergeyKanzhelev** [comment]
@dhaval24 there is no callback. Callback can be implemented by inspecting environment variables. This is how it works =(

File: web/src/test/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilterTests.java (hunk 2 above)
Comment (littleaj) [reply web/src/test/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilterTests.java]: `// region` is recognized by IntelliJ.

File: web/src/test/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilterTests.java (hunk 1 above)
Comment (littleaj) [reply web/src/test/java/com/microsoft/applicationinsights/web/internal/WebRequestTrackingFilterTests.java]: We can clean up consistency issue like this in a style related PR.

**Pull request merged**